java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinMapOperations.java

```diff
@@ -72,13 +72,16 @@ public static Term builtinMapLookup(Term map, Term key) {
     }
 
     public static Term update(Term map, Term key, Term value, TermContext context) {
-        if((map instanceof BuiltinMap) && ((BuiltinMap) map).isConcreteCollection() && key.isConcrete()) {
-            BuiltinMap.Builder builder = BuiltinMap.builder(context.global());
-            builder.put(key, value);
-            return updateAll(map, (BuiltinMap) builder.build(), context);
-        } else {
-            return  null;
+        if (map instanceof BuiltinMap) {
+            BuiltinMap bMap;
+            bMap = (BuiltinMap) map;
+            if((bMap.isConcreteCollection() && key.isConcrete()) || bMap.getEntries().containsKey(key)) {
```

Comment:
@ehildenb `isConcreteCollection()` does not mean that all the map entries are concrete.  It just says that there is no other variables or abstractions, like `(1 |-> 2  M:Map)` or `(1 |-> 2  tree(X))`.

Reply:
Ok, well I will come up with another fix, and make sure it passes/fails these tests correctly. Then will try out on KEVM master again and then ask you to try on KEVM Gnosis branch again.

Reply:
I'll add more tests later. Need to sit down and spend time to come up with exhaustive test cases.

Reply:
We don't need exhaustive tests. We just need any tests at all. Any test you want to write is a good test.